src/pages/ServerRoleSummary/NicMappingTab.js

```diff
@@ -38,9 +41,33 @@ class NicMappingTab extends Component {
       // Since many fields on the form can be edited at the same time, the validity of
       // each field is tracked separately within the detail row entry
       detailRows: undefined,
+      // check against nic-mappings in use when
+      // isUpdateMode
+      nicMappingsInUse: undefined,
+      loading: false,
+      errorMsg: undefined
     };
   }
 
+  componentDidMount() {
+    if(this.props.isUpdateMode) {
+      this.setState({loading: true});
+      // fetchJson(url, init, forceLogin, noCache)
+      fetchJson('/api/v2/model/nic_mappings_in_use', undefined, true, true)
```

Comment:
As we discussed in person, for consistency with the rest of the UI, it probably makes more sense to just use the values here from the internal model (the output of the config processor). This API only deals with the corner case where the config processor has been run but the ready deployment playbook has not been run; if in that situation we want to ignore the config processor output and instead look in the scratch dir, then we need to do that across the entirety of the day 2 UI for consistency. That would include all of the topology pages. It would probably make more sense to handle that centrally by modifying the `getInternalModel` to ignore the cp output and just pluck out the information from the scratch directory.

Reply:
@GarySmith per conversation, will update to call getInternalModel

Reply:
@GarySmith .starting working on the changes regarding calling getInternalModel...then I am thinking...getInternalModel still goes to backend to fetch the internal model and return a big object + I need to add UI process to extract the info I need..while calling get_nic_mapping_in_use returns a much smaller object which is ready to be consumed by UI.   I am wondering whether it makes sense to change to getInternalModel now.

Reply:
@guglo Keep in mind that the UI caches the internalModel, so it will only fetch a new one the first time after the config processor is run. So actually it would use less memory and less time to make use of the information that is already cached.

Reply:
@GarySmith to resolve the consistency...I can update the backend to not use ansible resources command.

Reply:
@guglo This function belongs in the UI, which usually has the model cached already. It does not make sense to go calculate this on the back end.

Reply:
@GarySmith  ok. will update.